doc/examples/reconst_ivim.py

```diff
@@ -227,6 +226,89 @@ def plot_map(raw_data, variable, limits, filename):
 plot_map(ivimfit.D_star, "D*", (0, 0.01), "perfusion_coeff.png")
 plot_map(ivimfit.D, "D", (0, 0.001), "diffusion_coeff.png")
 
+"""
+
+Next, we will fit the same model with a more refined optimization process with
+`fit_method='VarPro'` (for "Variable Projection"). The VarPro computes the
+IVIM parameters using the MIX approach [Farooq16]_. This algorithm uses three
+different optimizers. It starts with a differential evolution
+algorithm and fits the parameters in the power of exponentials. Then the fitted
+parameters in the first step are utilized to make a linear convex problem.
+Using a convex optimization, the volume fractions are determined. The last step
+is non-linear least-squares fitting on all the parameters. The results of the first
+and second optimizers are utilized as the initial values for the last step of
+the algorithm.
+
+As opposed to the `'LM'` fitting method, this approach does not need to set any
+thresholds on the bvals to differentiate between the perfusion
+(pseudo-diffusion) and diffusion portions and fits the parameters simultaneously.
+Making use of the three step optimization mentioned above increases the
+convergence basin for fitting the multi-exponential functions of
+microstructure models. This method has been described in further detail in
+[Fadnavis19]_ and [Farooq16]_.
+"""
+
+ivimmodel_vp = IvimModel(gtab, fit_method='VarPro')
+
+t1 = time.time()
+ivimfit_vp = ivimmodel_vp.fit(data_slice)
+t2 = time.time()
+total = t2 - t1
+
+"""
+Just like the `'LM'` fit method, `'VarPro'` creates a IvimFit object which
+contains the parameters of the model obtained after fitting. These are
+accessible through the `model_params` attribute of the IvimFit object.
+The parameters are arranged as a 4D array, corresponding to the spatial
+dimensions of the data, and the last dimension (of length 4)
+corresponding to the model parameters according to the following
+order : $\mathbf{S_{0}, f, D^*, D}$.
+"""
+
+i, j = 10, 10
+estimated_params = ivimfit_vp.model_params[i, j, :]
+print(estimated_params)
+
+"""
+Next, we plot the results relative to the model fit with `fit_method='VarPro'`.
+For this we will use the `predict` method of the IvimFit object
+to get the estimated signal.
+"""
+
+ivim_predict_vp = ivimfit_vp.predict(gtab)[i, j, :]
+plt.scatter(gtab.bvals, data_slice[i, j, :],
+            color="green", label="Actual signal")
+plt.plot(gtab.bvals, ivim_predict_vp, color="red",
+         label="Estimated Signal")
+plt.xlabel("bvalues")
+plt.ylabel("Signals")
+
+S0_est, f_est, D_star_est, D_est = estimated_params
+
+text_fit = """Estimated \n S0={:06.3f} f={:06.4f}\n
+            D*={:06.5f} D={:06.5f}""".format(S0_est, f_est, D_star_est, D_est)
+
+plt.text(0.65, 0.50, text_fit, horizontalalignment='center',
+         verticalalignment='center', transform=plt.gca().transAxes)
+plt.legend(loc='upper right')
+plt.savefig("ivim_voxel_plot.png")
```

Comment:
@arokem Can I just do `plt.close()` here so that the labels don't overlap as above? 

I think that should work as the next plots make use of the `plot_map()` helper function anyway!

Reply:
I think that a `plt.figure()` here should create a new figure and that's where `plot_map` would then operate. Does that make sense?

Reply:
@arokem on it! Sounds good!